enterprise/server/remote_execution/operation/operation.go

```diff
@@ -167,31 +240,38 @@ func (c *retryingClient) CloseAndRecv() (*repb.PublishOperationResponse, error)
 	return nil, status.UnknownError("CloseAndRecv: unknown error")
 }
 
+// TODO: make assemble() the public one and remove this.
 func Assemble(stage repb.ExecutionStage_Value, name string, r *digest.ResourceName, er *repb.ExecuteResponse) (*longrunning.Operation, error) {
 	if r == nil || er == nil {
 		return nil, status.FailedPreconditionError("digest or execute response are both required to assemble operation")
 	}
-	metadata, err := anypb.New(&repb.ExecuteOperationMetadata{
+	md := &repb.ExecuteOperationMetadata{
 		Stage:        stage,
 		ActionDigest: r.GetDigest(),
-	})
-	if err != nil {
-		return nil, err
 	}
-	operation := &longrunning.Operation{
-		Name:     name,
-		Metadata: metadata,
+	return assemble(name, md, er)
+}
+
+func assemble(name string, md *repb.ExecuteOperationMetadata, er *repb.ExecuteResponse) (*longrunning.Operation, error) {
+	op := &longrunning.Operation{
+		Name: name,
+		Done: md.GetStage() == repb.ExecutionStage_COMPLETED,
 	}
-	result, err := anypb.New(er)
-	if err != nil {
-		return nil, err
+	if md != nil {
+		mdAny, err := anypb.New(md)
+		if err != nil {
+			return nil, err
+		}
+		op.Metadata = mdAny
 	}
-	operation.Result = &longrunning.Operation_Response{Response: result}
-
-	if stage == repb.ExecutionStage_COMPLETED {
-		operation.Done = true
+	if er != nil {
```

Comment:
uber-nit: Could you rename `er`? It looks like a typo of `err`

Reply:
haha it does :laughing: done